core/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -140,17 +137,16 @@ public static ActiveShardCount parseString(final String str) {
      * Returns true iff the given cluster state's routing table contains enough active
      * shards to meet the required shard count represented by this instance.
      */
-    public boolean enoughShardsActive(final ClusterState clusterState, final String indexName, final Settings settings) {
-        Objects.requireNonNull(clusterState);
-        Objects.requireNonNull(indexName);
-        Objects.requireNonNull(settings);
+    public boolean enoughShardsActive(final ClusterState clusterState, final String indexName) {
         if (this == ActiveShardCount.NONE) {
             // not waiting for any active shards
             return true;
         }
+        final IndexMetaData indexMetaData = clusterState.metaData().index(indexName);
+        assert indexMetaData != null;
```

Comment:
Ok, so we will treat it as if all required shards have become active and we are done waiting.